testing/framework/TestCmd.py

```diff
@@ -1701,12 +1714,12 @@ def run(self, program=None,
         if self.verbose >= 2:
             write = sys.stdout.write
             write('============ STATUS: %d\n' % self.status)
-            out = self.stdout()
+            out = self.stdout() or ""
```

Comment:
curious, why default to "", instead of getting None? (assuming thats the default if no output?

Reply:
Because things then did string-y things with it. Some of the ones I changed were indeed safe, but there were also cases like: `stats = self.collect_stats(self.stdout())`; that method takes that argument and does a regex search on it.  I can back some of these out if it bothers you - it felt better to change it more consistently, but only some places strictly needed it.

Reply:
to add to this... I don't really see any reason why the two retrieval functions (stderr() and stdout()) should ever need to return `None`... an empty string should be fine to indicate there was no output on that stream.  But I figured that had a greater chance of disrupting something somewhere.